mbedtls/tests/async_session.rs

```diff
@@ -51,78 +105,111 @@ async fn client(conn: TcpStream, min_version: Version, max_version: Version, exp
                                                             // VerifyError
             Ok(())
         };
-        let mut config = Config::new(Endpoint::Client, Transport::Stream, Preset::Default);
-        config.set_rng(rng);
         config.set_verify_callback(verify_callback);
         config.set_ca_list(cacert, None);
-        config.set_min_version(min_version)?;
-        config.set_max_version(max_version)?;
-        let mut ctx = Context::new(Arc::new(config));
+    } else {
+        // for psk-based operation, only PSK required
+        config.set_psk(&[0x12, 0x34, 0x56, 0x78], "client")?;
+    }
+    let mut ctx = Context::new(Arc::new(config));
+    // For DTLS, timers are required to support retransmissions
+    if C::get_transport_type() == Transport::Datagram {
+        ctx.set_timer_callback(Box::new(Timer::new()));
+    }
 
-        match ctx.establish_async(conn, None).await {
-            Ok(()) => {
-                assert_eq!(ctx.version(), exp_version.unwrap());
-            }
-            Err(e) => {
-                match e {
-                    Error::SslBadHsProtocolVersion => {
-                        assert!(exp_version.is_none())
-                    }
-                    Error::SslFatalAlertMessage => {}
-                    e => panic!("Unexpected error {}", e),
-                };
-                return Ok(());
-            }
-        };
+    match ctx.establish_async(conn, None).await {
+        Ok(()) => {
+            assert_eq!(ctx.version(), exp_version.unwrap());
```

Comment:
Not an expert, but why is this an assert which panics on error not an Err? Is this the user of this library's fault?

Reply:
Nvm, potentially, didn't realize this was a test file.